tools/test-npm.sh

```diff
@@ -17,6 +17,19 @@ rm -rf test-npm
 # make a copy of deps/npm to run the tests on
 cp -r deps/npm test-npm
 
+
+## a hopefully temporary hack to use the local $NODE binary when calling node(1)
+# make a link to $NODE as test-npm/node
+ln -fs $NODE node
+# get the absolute path we are in -- see: http://stackoverflow.com/a/4774063/1279026
+pushd `dirname $0` > /dev/null
+PWD=`pwd -P`
+popd > /dev/null
+# export that link as preferred in the $PATH
+export PATH=$PWD:$PATH
+##
```

Comment:
Ok I can confirm this works on both OS X and Linux (Ubuntu 15), it just means we have an extra thing to clean up in the project root, as seen at the bottom of this file.